### PR TITLE
Slider tap target size

### PR DESCRIPTION
**Pull request by sladecapasso**
## Description

* This PR is to add MaterialTapTargetSize to Material Slider and Material Range Slider, to address issue #33795
* This includes:
  * Updating MaterialTapTargetSize to list Slider and Range Slider
  * Updating Material slider and range_slider to use MaterialTapTargetSize
  * Updating slider_test and range_slider_test debug fill properties for materialTapTargetie

## Related Issues

* #33795

## Tests

I added the following tests:

* DiagnosticsProperties

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/range_slider.dart [hunk 1]
```diff
@@ -332,11 +334,39 @@ class RangeSlider extends StatefulWidget {
...
+    super.debugFillProperties(properties);
+    properties.add(DoubleProperty('value1', values.start));
+    properties.add(DoubleProperty('value2', values.end));
+    properties.add(ObjectFlagProperty<ValueChanged<RangeValues>>('onChanged', onChanged, ifNull: 'disabled'));
+    properties.add(ObjectFlagProperty<ValueChanged<RangeValues>>.has('onChangeStart', onChangeStart));
+    properties.add(ObjectFlagProperty<ValueChanged<RangeValues>>.has('onChangeEnd', onChangeEnd));
+    properties.add(DoubleProperty('min', min));
+    properties.add(DoubleProperty('max', max));
+    properties.add(IntProperty('divisions', divisions));
+    properties.add(StringProperty('label1', labels?.start));
```
Comment (clocksmith) [inline packages/flutter/lib/src/material/range_slider.dart]: call these start label and end label, or better yet, can you use the toString method on RangeLabels?

File: packages/flutter/lib/src/material/range_slider.dart [hunk 2]
```diff
@@ -332,11 +334,39 @@ class RangeSlider extends StatefulWidget {
...
   // Touch width for the tap boundary of the slider thumbs.
   static const double _minTouchTargetWidth = kMinInteractiveDimension;
 
   @override
   _RangeSliderState createState() => _RangeSliderState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DoubleProperty('value1', values.start));
```
Comment (clocksmith) [inline packages/flutter/lib/src/material/range_slider.dart]: start value and end value, or the toString method on RangeValues

**Review by clocksmith — CHANGES_REQUESTED** [review]
I would split this change up with range slider debug fill properties.

The material tap target size change should probably be separate. It should also be consumed by the Slider and RangeSlider.

File: packages/flutter/test/material/range_slider_test.dart [hunk 3]
```diff
@@ -1533,5 +1533,42 @@ void main() {
 
     await gesture.up();
   });
+
+  testWidgets('Range Slider implements debugFillProperties', (WidgetTester tester) async {
```
Comment (gspencergoog) [inline packages/flutter/test/material/range_slider_test.dart]: This is a fine test of the `debugFillProperties` method, but there will also need to be a test of the tap target size itself, by simulating taps in various places to show that it covers the right region.

**Review by gspencergoog — COMMENTED** [review]
@sladecapasso, are you still interested in working on this PR?  If so, it will need to have the file conflicts addressed, and as Anthony mentions, it could be split into two separate PRs.

File: packages/flutter/test/material/range_slider_test.dart (hunk 3 above)
Comment (sladecapasso) [reply packages/flutter/test/material/range_slider_test.dart]: It's been deprioritized to work on something else, but I'd like to come back to it.

Thank you for the feedback!

**Comment by gspencergoog** [comment]
OK, we're going to close this PR for now, then.  Please feel free to re-open it or create a new PR when you pick it back up again.

Thanks for helping!

**Pull request closed without merging**